modules/scheduler/gke-cluster/main.tf

```diff
@@ -177,6 +177,10 @@ resource "google_container_cluster" "gke_cluster" {
     ignore_changes = [
       node_config
     ]
+    precondition {
```

Comment:
Can we have a similar precondition for the gke-node-pool module as well? Probably you will need to use a google_container_cluster data source.